src/DAMA/DoctrineTestBundle/Doctrine/DBAL/StaticConnectionFactory.php

```diff
@@ -46,7 +46,13 @@ public function createConnection(array $params, Configuration $config = null, Ev
 
         if (StaticDriver::isKeepStaticConnections()) {
             // The underlying connection already has a transaction started.
-            // So we start it as well on this connection so the internal state ($_transactionNestingLevel) is in sync with the underlying connection.
+            // Make sure we use savepoints to be able to easily roll-back nested transactions
+            if ($connection->getDriver()->getDatabasePlatform()->supportsSavepoints()) {
+                $connection->setNestTransactionsWithSavepoints(true);
```

Comment:
@garak @Jean85 are you actively using this bundle? If so can you confirm that this does not break any of your testsuites?

I tested it with multiple of my projects including mysql, postgres and sqlite databases and for me it works fine.

Reply:
Yes I'm using it, but only with MySQL. I'll try it as soon as I can.

Reply:
@Jean85 any chance to test this?

Reply:
I would like to confirm it does not break things for you before releasing this :blush: